client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -144,6 +146,19 @@ public class TrinoConnection
         timeZoneId.set(ZoneId.systemDefault());
         locale.set(Locale.getDefault());
         sessionProperties.putAll(uri.getSessionProperties());
+
+        if (uri.isUseLegacyPreparedStatements().isPresent()) {
+            this.useLegacyPreparedStatements = uri.isUseLegacyPreparedStatements().get();
+        }
+        else {
+            try (Statement stmt = createStatement()) {
```

Comment:
Calling instance methods in a constructor can be dangerous, because the callee is an uninitialized object.  Also, taking the hit for every connection may be undesirable.  We should consider caching whether this capability is supported by a Trino instance at the class level.  Or given the fact that support for `EXECUTE IMMEDIATE` has been available for a while now, perhaps we skip detection and turn it on by default.  @hashhar , @findepi WDYT?

Reply:
Since this is a fairly new feature in Trino, we should make this disabled by default. Users can opt-in, and we can make this a default in the future after it is better tested.

I don't think it makes sense to test this behavior, since the entire purpose is to reduce round trips. If the user is on an old version of Trino, they can disable this behavior in the driver.

Reply:
Fair enough, code changed accordingly